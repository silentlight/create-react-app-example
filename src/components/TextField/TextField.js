import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class TextField extends PureComponent {
  render() {
    const { input, meta: { touched, error, warning } } = this.props;
    const { label, type } = this.props;

    return (
      <div>
        <label>
          {label}
        </label>
        <div>
          <input {...input} placeholder={label} type={type} />
          {touched &&
          ((error &&
          <p className="warning">
            {error}
          </p>) ||
          (warning &&
          <p className="error">
            {warning}
          </p>))}
        </div>
      </div>
    );
  }
}

TextField.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.string,
    warning: PropTypes.string,
  }),
};

TextField.defaultProps = {
  type: 'text'
};

export default TextField;
