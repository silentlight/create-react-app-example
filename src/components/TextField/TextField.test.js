import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TextField from './TextField';

const createComponent = (props) => {
  return shallow(<TextField {...props} />);
};

describe('TextField', () => {

  const props = {
    input: {
      name: 'Test',
    },
    label: 'Test field',
    meta: {
      touched: false,
      error: null,
      warning: null,
    }
  };

  let component = createComponent(props);

  it('renders and matches the snapshot', () => {
    const tree = renderer.create(<TextField {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has label', () => {
    expect(component.find('label').text()).toEqual(props.label);
  });

  it('has correct default type', () => {
    expect(component.find('input').prop('type')).toEqual('text');
  });

  it('has correct explicit type', () => {
    props.type = 'password';

    component = createComponent(props);

    expect(component.find('input').prop('type')).toEqual(props.type);
  });

  it('displays warning', () => {
    props.meta.touched = true;
    props.meta.warning = 'Warning exemple';

    component = createComponent(props);

    expect(component.html()).toContain(props.meta.warning);
  });

  it('displays error', () => {
    props.meta.touched = true;
    props.meta.error = 'Error exemple';

    component = createComponent(props);

    expect(component.html()).toContain(props.meta.error);
  });
});