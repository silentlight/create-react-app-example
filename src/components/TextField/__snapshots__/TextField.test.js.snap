// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders and matches the snapshot 1`] = `
<div>
  <label>
    Test field
  </label>
  <div>
    <input
      name="Test"
      placeholder="Test field"
      type="text"
    />
  </div>
</div>
`;
